front-end/src/screens/MenuBar/index.tsx

```diff
@@ -33,68 +35,210 @@ const MenuBar = ({ className } : Props): JSX.Element => {
 		logoutMutation();
 	};
 
+	// Menu Items
+	const contentItems = [
+		{ content:'Discussions', icon:'comments', key:'discussions', to:'/discussions' },
+		{ content: 'Proposals', icon:'file alternate', key:'proposals', to:'/proposals' }
+	];
+
+	const loggedOutItems = [
+		{ content:'Login', icon:'sign in', key:'login', to:'/login' },
+		{ content: 'Sign-up', icon:'plus circle', key:'signup', to:'/signup' }
+	];
+
 	const userMenu = <><Icon name='user circle' inverted />{username}</>;
 	const caretIcon = <Icon name='caret down' inverted />;
 
+	// Mobile Sidebar
+	const [menuVisible, setMenuVisible] = useState(false);
+	const [pushableHeight, setPushableHeight] = useState('0rem');
+
+	const handleToggle = () => {
+		menuVisible ? setMenuVisible(false) : setMenuVisible(true);
+		menuVisible ? setPushableHeight('0rem'): setPushableHeight('100%');
+	};
+
+	const handleClose = () => {
+		setMenuVisible(false);
+		setPushableHeight('0rem');
+	};
+
 	return (
-		<Menu className={className} stackable inverted borderless>
-			<Menu.Item as={Link} to="/" id='title'>Polkassembly</Menu.Item>
-			<Menu.Item as={Link} to="/discussions" id='title'>Discussions</Menu.Item>
-			<Menu.Item as={Link} to="/proposals" id='title'>Proposals</Menu.Item>
-			<Menu.Menu position="right">
-				{username
-					? <>
-						<Dropdown trigger={userMenu} icon={caretIcon} item={true}>
-							<Dropdown.Menu>
-								<Menu.Item as={Link} to="/settings">Settings</Menu.Item>
-								<Menu.Item onClick={handleLogout}>Logout</Menu.Item>
-							</Dropdown.Menu>
-						</Dropdown>
-					</>
-					: <>
-						<Menu.Item as={Link} to="/login">Login</Menu.Item >
-						<Menu.Item as={Link} to="/signup">Sign-up</Menu.Item >
-					</>
-				}
-			</Menu.Menu>
-		</Menu>
+		<>
+			<Responsive maxWidth={Responsive.onlyTablet.maxWidth}>
+				<Menu className={className} inverted widths={2} id='menubar'>
+					<Menu.Item onClick={handleClose} as={Link} to="/" id='title'>
+						Polkassembly
+					</Menu.Item>
+					<Menu.Item onClick={handleToggle} id='rightmenu'>
+						{!menuVisible ? <Icon name="sidebar" /> : <Icon name="times" />}
+					</Menu.Item>
+				</Menu>
+				<Sidebar.Pushable className={className} style={{ height:pushableHeight }}>
+					<Sidebar
+						as={Menu}
+						direction='top'
+						icon="labeled"
+						inverted
+						stackable
+						vertical
+						visible={menuVisible}
+					>
+						{contentItems.map(item => <Menu.Item as={Link} key={item.key} onClick={handleClose} {...item} />)}
+						{username
+							?
+							<>
+								<Menu.Item as={Link} to="/settings" onClick={handleClose}><Icon name="cog" />Settings</Menu.Item>
+								<Menu.Item onClick={handleLogout}><Icon name="sign-out" />Logout</Menu.Item>
```

Comment:
These menus are duplicated afaics.

Reply:
I wasn't quite sure with the onClick prop, but seems to work fine like
```
const loggedInItems = [
		{ content:'Settings', icon:'cog', key:'settings', to:'/settings' },
		{ content: 'Logout', icon:'sign-out', key:'signout', onClick: handleLogout, to:'/' }
	];
```